orchagent/pfc_detect_mellanox.lua

```diff
@@ -71,6 +71,8 @@ for i = n, 1, -1 do
                         -- DEBUG CODE END.
                         (occupancy_bytes == 0 and packets - packets_last == 0 and (pfc_duration - pfc_duration_last) > poll_time * 0.8) then
                         if time_left <= poll_time then
+                            redis.call('HDEL', counters_table_name .. ':' .. port_id, pfc_rx_pkt_key .. '_last')
```

Comment:
Need to understand how the change impacts the restore logic because restore relies on SAI_PORT_STAT_PFC_?_RX_PKTS_last, which is now reset on each storm detection. Because SAI_PORT_STAT_PFC_?_RX_PKTS_last is reset at a storm detection, the immediate next polling interval following the storm detection, the restore script sees pfc_rx_packets_last == 0, and skips the restore logic. https://github.com/Azure/sonic-swss/blob/6007e7f68cc103784208f69f8e6a0f12d4f2a193/orchagent/pfc_restore.lua#L42

So if there exists a situation that the storm restore is detected at the next polling interval of the storm detection occurrence, the restore signal is now delayed to 2 * polling intervals. Does this situation happen in realistic life, or its occurrence is just theorical?

Reply:
restoration script resets this field but will do it in proper time after orchagent reacted to the storm
```
if not ... pfc_wd_status ~= 'operational' ...
    ...
    redis.call('HSET', counters_table_name .. ':' .. port_id, pfc_rx_pkt_key .. '_last', pfc_rx_packets)
end
```

Reply:
If _RX_PKTS_last and _RX_PAUSE_DURATION_last is hdeleted at a storm signal, the earliest restore signal will be two polling intervals away from the time a detect signal is published.